File: arrays-hashing/medium/longest-consecutive-sequence/sol.py
# https://leetcode.com/problems/longest-consecutive-sequence/description/

from typing import List


class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        mySet = set(nums)
        ans = 0

        for n in mySet:
            # check start of sequence
            if (n - 1) not in mySet:
                length = 1

                while (n + length) in mySet:
                    length += 1

                ans = max(ans, length)

        return ans

File: arrays-hashing/medium/longest-consecutive-sequence/sol.cs
// https://leetcode.com/problems/longest-consecutive-sequence/description/

public class Solution
{
    public int LongestConsecutive(int[] nums)
    {
        HashSet<int> mySet = new HashSet<int>(nums);
        int ans = 0;

        foreach (int n in mySet)
        {
            // check start of sequence
            if (!mySet.Contains(n - 1))
            {
                int length = 1;

                while (mySet.Contains(n + length))
                {
                    length += 1;
                }

                ans = Math.Max(ans, length);
            }
        }

        return ans;
    }
}
